### PR TITLE
Pass form data to AJAX callback

### PR DESCRIPTION
**Pull request by katzmo**

### CODE REVIEW

File: src/Component.php [hunk 1]
```diff
@@ -362,7 +362,11 @@ public function executeAjaxCallback(\PaymentMethod $method, &$form, &$form_state
     }
     $payment = $this->payment;
     $payment->method = $method;
-    $result = $method->controller->ajaxCallback($payment, $form, $form_state);
+    // Pass only controller specific form data to the callback.
```
Comment (torotil) [inline src/Component.php]: This comment is only helpful when you take into account that the whole form element was passed in the commmit before. As a description for the current behaviour I find it rather confusing. I would either leave it or change it to something like:

> Pass method form element to the callback.

File: src/Component.php [hunk 2]
```diff
@@ -362,7 +362,11 @@ public function executeAjaxCallback(\PaymentMethod $method, &$form, &$form_state
...
+    $webform = new Webform($form['#node']);
+    $element = &drupal_array_get_nested_value($form['submitted'], $this->parents($webform));
+    $element = &$element['payment_method_all_forms'][$method->pmid];
```
Comment (torotil) [inline src/Component.php]: I would avoid the references here: The callback should not edit the element and if it does this shouldn’t have a consequence.

**Review by torotil — CHANGES_REQUESTED** [review]
Thanks for preparing the change. Sorry to be so picky here, but with code this complex I fear I must.

File: src/Component.php (hunk 1 above)
Comment (katzmo) [reply src/Component.php]: I'm taking into account that the whole $form (presumably) was passed as argument and now we filter out the relevant part. But happy to remove the comment entirely :)

File: src/Component.php (hunk 2 above)
Comment (katzmo) [reply src/Component.php]: Have updated the last commit.

File: src/Component.php (hunk 1 above)
Comment (torotil) [reply src/Component.php]: Yes, the whole `$form` was passed — in the previous commit of this very PR. This branch introduces two new arguments.

**Review by torotil — APPROVED** [review]

**Pull request merged**